### PR TITLE
front: hard code composition codes

### LINKED ISSUE #8729: Show only composition codes for Freight trains

### LINKED ISSUE #9311: front: hard code composition codes used for the freight companies
### Description and goal

Hard code the composition codes used for the freight companies.

The list:

```
HLP
MA100
MA80
MA90
ME100
ME120
ME140
MV160
MVGV
Marchandise
Marchandise - MA100
Marchandise - MA80
Marchandise - MA90
Messagerie - ME100
Messagerie - ME120
Messagerie - ME140
Messagerie - MVGV
```

### Acceptance criteria

- [ ] the stdcm interface is easier to use for the freight companies

### PR DESCRIPTION
**Pull request by theocrsb**
closes #9311 
closes #8729

### CODE REVIEW

**Review by Akctarus — APPROVED** [review]
Tested, LGTM

File: front/src/assets/compositionCode/compositionCodes.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const COMPOSITION_CODES = [
```
Comment (emersion) [inline front/src/assets/compositionCode/compositionCodes.ts]: Should we somehow name this file in a way that makes it clear it's for STDCM specifically?

Comment (theocrsb) [reply front/src/assets/compositionCode/compositionCodes.ts]: I think this list could be used elsewhere

Comment (emersion) [reply front/src/assets/compositionCode/compositionCodes.ts]: The list is very specific to STDCM I believe?

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, good job !

File: front/src/assets/compositionCode/compositionCodes.ts (hunk 1 above)
Comment (SharglutDev) [reply front/src/assets/compositionCode/compositionCodes.ts]: Yes it's for stdcm

Comment (SharglutDev) [reply front/src/assets/compositionCode/compositionCodes.ts]: we could rename it `stdcmCompositionCodes` ?

Comment (theocrsb) [reply front/src/assets/compositionCode/compositionCodes.ts]: I can, @emersion, you think it's better ?

Comment (emersion) [reply front/src/assets/compositionCode/compositionCodes.ts]: Yes, I think putting "stdcm" in the filename is a good idea.

Comment (theocrsb) [reply front/src/assets/compositionCode/compositionCodes.ts]: Done

File: front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts [hunk 2]
```diff
@@ -4,13 +4,14 @@ import { compact, concat, uniq } from 'lodash';
 import { useTranslation } from 'react-i18next';
 import { useSelector } from 'react-redux';
 
+import COMPOSITION_CODES from 'assets/compositionCode/compositionCodes';
```
Comment (emersion) [inline front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts]: This import seems to still reference the old file?

File: front/src/assets/compositionCode/stdcmCompositionCodes.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const COMPOSITION_CODES = [
```
Comment (emersion) [inline front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Ah, one more thing: do we really want to put this file in `assets/`? Assets are copied over in a directory when building the web app, but this is just a .ts file imported like any other.

File: front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts (hunk 2 above)
Comment (theocrsb) [reply front/src/common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector.ts]: Done

File: front/src/assets/compositionCode/stdcmCompositionCodes.ts (hunk 3 above)
Comment (theocrsb) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: It's true that he could be somewhere else. In `src/application/stdcm/const.ts` perhaps?

Comment (SharglutDev) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: I think @theocrsb based this with @clarani's PR for hard coded stdcm rolling stocks.

Comment (theocrsb) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Yes

Comment (emersion) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Yeah, somewhere in `src/application/stdcm/` sounds good to me.

Comment (theocrsb) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Even in `const` ?

Comment (emersion) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Either `src/application/stdcm/consts.ts` (we use the plural form everywhere else), or `src/application/stdcm/consts/compositionCodes.ts` if it's deemed too much data (with the upcoming rolling stock list).

Comment (theocrsb) [reply front/src/assets/compositionCode/stdcmCompositionCodes.ts]: Done

**Review by emersion — APPROVED** [review]
Sorry for the back-and-forth, LGTM :)

**Pull request merged**